src/python/pants/backend/go/target_types.py

```diff
@@ -140,18 +146,41 @@ class GoExternalPackageTarget(Target):
 # -----------------------------------------------------------------------------------------------
 
 
-class GoBinaryMainAddress(StringField):
+class GoBinaryMainPackageField(StringField, AsyncFieldMixin):
     alias = "main"
-    required = True
-    help = "Address of the main Go package for this binary."
+    help = (
+        "Address of the `go_package` with the `main` for this binary.\n\n"
+        "If left off, will default to the `go_package` in the same directory as this target's "
```

Comment:
"left off" is an idiomatic phrasing. "If not specified" might be a better phrasing.